package-dev/Plugins/macOS/SentryNativeBridge.m

```diff
@@ -0,0 +1,220 @@
+#import <Foundation/Foundation.h>
+#include <dlfcn.h>
+
+static int loadStatus = -1; // unitialized
+
+static Class SentrySDK;
+static Class SentryScope;
+static Class SentryBreadcrumb;
+static Class SentryUser;
+
+#define LOAD_CLASS_OR_BREAK(name)                                                                  \
+    name = (__bridge Class)dlsym(dylib, "OBJC_CLASS_$_" #name);                                    \
+    if (!name) {                                                                                   \
+        NSLog(@"Sentry (native bridge): Couldn't load %@ class from the dynamic library", name);   \
+        break;                                                                                     \
+    }
+
+// Returns (bool): 0 on failure, 1 on success
+// WARNING: you may only call other Sentry* functions AFTER calling this AND only if it returned "1"
+int SentryNativeBridgeLoadLibrary()
+{
+    if (loadStatus == -1) {
+        loadStatus = 0; // init to "error"
+        do {
+            void *dylib = dlopen("@executable_path/../PlugIns/Sentry.dylib", RTLD_LAZY);
+            if (!dylib) {
+                NSLog(@"Sentry (native bridge): Couldn't load Sentry.dylib - dlopen() failed");
+                break;
+            }
+
+            LOAD_CLASS_OR_BREAK(SentrySDK)
+            LOAD_CLASS_OR_BREAK(SentryScope)
+            LOAD_CLASS_OR_BREAK(SentryBreadcrumb)
+            LOAD_CLASS_OR_BREAK(SentryUser)
+
+            // everything above passed - mark as successfully loaded
+            loadStatus = 1;
+        } while (false);
+    }
+    return loadStatus;
+}
+
+const void *SentryNativeBridgeOptionsNew()
+{
+    return CFBridgingRetain([[NSMutableDictionary alloc] init]);
+}
+
+void SentryNativeBridgeOptionsSetString(const void *options, const char *name, const char *value)
+{
+    NSMutableDictionary *dictOptions = (__bridge NSMutableDictionary *)options;
+    dictOptions[[NSString stringWithUTF8String:name]] = [NSString stringWithUTF8String:value];
+}
+
+void SentryNativeBridgeOptionsSetInt(const void *options, const char *name, int32_t value)
+{
+    NSMutableDictionary *dictOptions = (__bridge NSMutableDictionary *)options;
+    dictOptions[[NSString stringWithUTF8String:name]] = [NSNumber numberWithInt:value];
+}
+
+void SentryNativeBridgeStartWithOptions(const void *options)
+{
+    NSMutableDictionary *dictOptions = (__bridge_transfer NSMutableDictionary *)options;
+    [SentrySDK performSelector:@selector(startWithOptions:) withObject:dictOptions];
+}
+
+void SentryConfigureScope(void (^callback)(id))
+{
+    @try {
+        [SentrySDK performSelector:@selector(configureScope:) withObject:callback];
+    } @catch (NSException *exception) {
+        NSLog(@"Sentry (bridge): failed to configure scope: %@", exception.reason);
+    }
```

Comment:
`m`: Why do we wrap these calls with try catch?

Reply:
setValue forKey: may throw if the property is not found; same for performSelector I think - even though this shouldn't happen, better not take the chance of letting an unhandled exception while setting error info - it would just crash the app immediately.